packages/components/src/badge/index.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { info, caution, error, published } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import type { BadgeProps } from './types';
+import Icon from '../icon';
+
+function Badge( {
+	className,
+	as: Component = 'div',
+	context = 'neutral',
+	children,
+	...props
+}: BadgeProps ) {
+	/**
+	 * Returns an icon based on the badge context.
+	 *
+	 * @return {JSX.Element | null} The corresponding icon for the provided context.
+	 */
+	function contextBasedIcon(): JSX.Element | null {
```

Comment:
```suggestion
	function contextBasedIcon() {
```

I'm actually fine with removing this explicit annotation as well (because TS can infer that).